File: php/features.php
<?php
declare(strict_types=1);

// YadorePublisher SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class YadorePublisherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new YadorePublisherBaseFeature();
            case "test":
                return new YadorePublisherTestFeature();
            default:
                return new YadorePublisherBaseFeature();
        }
    }
}

File: py/yadorepublisher_sdk/features.py
# YadorePublisher SDK feature factory

from yadorepublisher_sdk.feature.base_feature import YadorePublisherBaseFeature
from yadorepublisher_sdk.feature.test_feature import YadorePublisherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YadorePublisherBaseFeature(),
        "test": lambda: YadorePublisherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
